runtime/gc_include/ObjectAllocationAPI.hpp

```diff
@@ -173,6 +173,11 @@ class MM_ObjectAllocationAPI
 						/* Do not zero the TLH if it is already zero's */
 						initializeSlots = initializeSlots && _initializeSlotsOnTLHAllocate;
 #endif /* J9VM_GC_BATCH_CLEAR_TLH */
+#if defined(J9VM_GC_REALTIME)
+						MM_GCRememberedSetFragment *fragment =  &currentThread->sATBBarrierRememberedSetFragment;
```

Comment:
@amicic you the following comments about the code:
> It's probably not required, since the last condition (0 != parent->globalFragmentIndex) is not ever met when SATB barrier is not enabled
> But we can think if it is easy to cash if SATB is active (but keep it up to date as GC cycles start end stop), in this class instead of fetching it through a few indirect references for each object allocation...

Will look into it

Reply:
If we're truly following metronome example here then `initializeSlots` should always be true for SATB config regardless of it's active or not. 

I can't think of a reason metronome has this unconditionally true, but the comment indicates it's necessary given the barrier (even if its not active).

Reply:
initializeSlots is not related to SATB. 
It's about if TLH is batch cleared or not (what is platform specific)

Reply:
```C++
if (cellSize <= ((uintptr_t) cacheEntry->top - (uintptr_t) cacheEntry->current)) {
	objectHeader = (j9object_t)cacheEntry->current;
	cacheEntry->current = (uintptr_t *) ((uintptr_t) cacheEntry->current + cellSize);
	/* The metronome pre write barrier might scan this object - always zero it */
	initializeSlots = true;
} 
```

Reply:
When a TLH is not batch cleared, would `initializeSlots` always be true?

Reply:
Yes, if TLH is not batch cleared, initializeSlots should be true, but Metronome does not use traditional TLHs, so (from looking at the code at _TYPE_SEGREGATED case) batch clearing does not seem be a consideration for Metronome, which might mean that platforms that normally support batch clearing of TLH (like AIX, where Metronome is indeed supported) would not support batch clearing of Metronome alloc caches. So we would have to clear the object anyway (it's just a general requirement, since object constructor may not explicitly zero integers or null ref slots)

Why is this necessary for barrier purposes (if SATB is active, if that is relevant, although the non-conditional code suggests it's not), I'm not completely sure, either. We certainly do not scan objects in the barrier (not directly, as the comment may suggest), but we do read slots (and act if slots are non-NULL, so that we may push them for scanning). 
But is it really possible that non-constructed object is subject to mutation of their references (or in other the words if the object address is exposed to mutators before construction is complete)... I can't remember, now.

Reply:
Thanks for that.

> But is it really possible that non-constructed object is subject to mutation of their references (or in other the words if the object address is exposed to mutators before construction is complete)... I can't remember, now.

I did see issue without initialization, specifically by code around `cloneObject`. It lead to objs with garbage fields being pushed for scanning and leading to crash